src/Microsoft.AspNet.FileProviders.Physical/PhysicalFileProvider.cs

```diff
@@ -199,7 +203,7 @@ public IChangeToken Watch(string filter)
             return _filesWatcher.CreateFileChangeToken(filter);
         }
 
-        private class PhysicalFileInfo : IFileInfo
+        internal class PhysicalFileInfo : IFileInfo
```

Comment:
Make these `public`. No reason for these types to be internal. Also move it to a separate file.